ui/components/multichain/pages/all-connections/all-connections.js

```diff
@@ -0,0 +1,183 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { Header, Page } from '../page';
+import {
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+  Text,
+} from '../../../component-library';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import {
+  BackgroundColor,
+  Color,
+  TextAlign,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import { DEFAULT_ROUTE } from '../../../../helpers/constants/routes';
+import {
+  getAllConnectedAccounts,
+  getConnectedSubjectsForAllAddresses,
+  getSnapsList,
+} from '../../../../selectors';
+import { Tab, Tabs } from '../../../ui/tabs';
+import { ConnectionListItem } from './connection-list-item';
+
+export const AllConnections = () => {
+  const t = useI18nContext();
+  const history = useHistory();
+  const TABS_THRESHOLD = 5;
+  let totalConnections = 0;
+  const connectedSubjectsForAllAddresses = useSelector(
+    getConnectedSubjectsForAllAddresses,
+  );
+  const connectedAddresses = useSelector(getAllConnectedAccounts);
+  const connectedSiteData = {};
+  connectedAddresses.forEach((connectedAddress) => {
+    connectedSubjectsForAllAddresses[connectedAddress].forEach((app) => {
+      if (!connectedSiteData[app.origin]) {
+        connectedSiteData[app.origin] = { ...app, addresses: [] };
+      }
+      connectedSiteData[app.origin].addresses.push(connectedAddress);
+    });
+  });
```

Comment:
Can we move them to `useMemo` to cache the results? you can find the documentation [here](https://react.dev/reference/react/useMemo) :)

Reply:
@DDDDDanica, thanks for the great feedback, but I'll be doing the performance refactors (tsx, hooks) in future PRs. 
Happy to sit down with you someday to refactor though, really awesome suggestions.
I'll definitely keep these tips in mind during future PRs!

Reply:
deal!